homework/index.js

```diff
@@ -29,22 +29,74 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function createTag(name, options = {}) {
+    const elem = document.createElement(name);
+    Object.entries(options).forEach(([key, value]) => {
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  function buildTableRow(title, parent, content) {
+    const row = createAndAppend(`tr`, parent);
+    createAndAppend(`th`, row, { text: `${title}:` });
+    let data;
+    if (typeof content === `string` || typeof content === `number`) {
+      createAndAppend(`td`, row, { text: `${content}` });
+    } else {
+      data = createAndAppend(`td`, row);
+      data.appendChild(content);
+    }
+  }
```

Comment:
This function and well as the previous (`createTag`) function is a good attempt at making your code DRY and extract code segments from a larger function into a couple of smaller functions. However, the resulting code is unnecessarily complicated, because of the `if` statement. In the `buildTableRow` function (good name 👍), you are trying to deal with exceptions, where content is not a text string or a number but an object. This makes this function difficult to understand. It is better to let the caller of this function handle the exceptions. You could then simplify this function to:

```js
function buildTableRow(title, parent, content) {
  const row = createAndAppend(`tr`, parent);
  createAndAppend(`th`, row, { text: `${title}:` });
  createAndAppend(`td`, row, { text: `${content}` });
  return row;
}
```

Note the addition of `return row;`. This allows the caller of the function to further work with the returned `tr`.